File: 704-binary-search/704-binary-search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        
        left, right = 0, len(nums) - 1

        while left <= right:
            
            pivot = left + (right-left)//2
                
            if nums[pivot] == target:
                return pivot
            
            if nums[pivot] < target:
                left = pivot + 1
            else:
                right = pivot - 1
                
                
            
        return -1

File: 0704-binary-search/704-binary-search.go
func search(nums []int, target int) int {
	left, right := 0, len(nums) - 1

	for right >= left {
		mid := (left + right) / 2

		if target == nums[mid] {
			return mid
		} else if target > nums[mid] {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	return -1
}